configuration/azure/appconfig/appconfig.go

```diff
@@ -248,9 +266,103 @@ func (r *ConfigurationStore) getLabelFromMetadata(metadata map[string]string) *s
 }
 
 func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	return "", fmt.Errorf("Subscribe is not implemented by this configuration store")
+	subscribeID := uuid.New().String()
+	stop := make(chan struct{})
+	r.subscribeStopChanMap.Store(subscribeID, stop)
+	sentinelKey := r.getSentinelKeyFromMetadata(req.Metadata)
+	if sentinelKey == "" {
+		return "", fmt.Errorf("sentinel key is not provided in metadata")
+	}
+	actualKeyToSubscribeSlice := make([]string, 0)
+	for _, key := range req.Keys {
+		_, found := r.keysToSubIDMap[key]
+		if !found {
+			actualKeyToSubscribeSlice = append(actualKeyToSubscribeSlice, key)
+			r.keysToSubIDMap[key] = subscribeID
+		}
+	}
+	if len(actualKeyToSubscribeSlice) == 0 {
+		return "", fmt.Errorf("all provided keys are already subscribed")
+	}
+	value, found := r.subIDToKeyListMap[subscribeID]
+	if found {
+		value = append(value, actualKeyToSubscribeSlice...)
+		r.subIDToKeyListMap[subscribeID] = value
+	} else {
+		r.subIDToKeyListMap[subscribeID] = actualKeyToSubscribeSlice
+	}
+
+	req.Keys = actualKeyToSubscribeSlice
+	go r.doSubscribe(ctx, req, handler, sentinelKey, subscribeID, stop)
```

Comment:
Can we pass in keys and metadata separately instead of modifying req and then passing in that?

Reply:
No need to modify the req any ore